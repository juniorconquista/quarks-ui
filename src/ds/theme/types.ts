import {
    ColorProps,
    FlexboxProps,
    LayoutProps,
    SpaceProps,
    PositionProps,
    BorderProps,
    GridProps,
} from 'styled-system';

export type DefaultStyleProps = BorderProps &
    ColorProps &
    FlexboxProps &
    LayoutProps &
    PositionProps &
    SpaceProps &
    GridProps & {
        // for some reason this isn't included; this is for replacing the 'base' component
        // as?: React.ComponentType;
    };
