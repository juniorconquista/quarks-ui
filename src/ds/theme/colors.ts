export default {
    dark: false,
    primary: {
        default: '#F26522',
        dark: '#d14b0c',
        light: '#f69263',
        contrast: '#ffffff',
    },
    secondary: {
        default: '#372E2A',
        dark: '#2c2522',
        light: '#453935',
        contrast: '#1B2B3A',
    },
    danger: {
        default: '#ff6961',
        dark: '#ff261b',
        light: '#ffbdb9',
        contrast: '#ffffff',
    },
    success: {
        default: '#0E9d51',
        dark: '#0a753c',
        light: '#10bc60',
        contrast: '#ffffff',
    },
    warning: {
        default: '#ffb143',
        dark: '#e4902c',
        light: '#ffd868',
        contrast: '#ffffff',
    },
    info: {
        default: '#0ea7a7',
        dark: '#0a7676',
        light: '#11d4d4',
        contrast: '#ffffff',
    },
    grays: {
        default: '#9ba8b2',
        dark: '#4b4b4b',
        light: '#eeeeef',
        contrast: '#f5f5f5',
    },
    background: {
        default: '#ffffff',
        dark: '#F26522',
    },
    text: {
        light: '#ffffff',
        dark: '#1B2b3a',
    },
    disabled: '#9398a6',
    primaries: ['#d7edf4'],
    whites: ['#eeeeef', '#f2fbfe'],
    blacks: ['#35363a'],
    hovers: ['#f5f7f7'],
};
