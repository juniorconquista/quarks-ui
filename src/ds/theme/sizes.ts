export const breakpoints = ['0rem', '37.5rem', '62rem'];

export const space = {
    none: '0',
    xxxs: '0.285rem',
    xxs: '0.571rem',
    xs: '0.857rem',
    sm: '1.142rem',
    md: '1.714rem',
    lg: '2.5rem',
    xl: '3.428rem',
    xxl: '6.857rem',
};

export const fontSizes = [
    '0.5rem',
    '0.85rem',
    '1rem',
    '1.5rem',
    '1.8rem',
    '2rem',
];

export const buttonSizes = {
    medium: {
        paddingTop: 8,
        paddingBottom: 8,
        paddingRight: 16,
        paddingLeft: 16,
        borderRadius: 16,
    },
    small: {
        paddingTop: 4,
        paddingBottom: 4,
        paddingRight: 16,
        paddingLeft: 16,
        borderRadius: 16,
    },
};

export const radii = {
    none: '0',
    sm: '0.285rem',
    md: '2.25rem',
    lg: '1.142rem',
    xl: '1.714rem',
};
