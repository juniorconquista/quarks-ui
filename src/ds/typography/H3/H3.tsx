import styled from 'styled-components';
import { variant, VariantArgs, color, typography, space } from 'styled-system';
import { H3StyledProps } from './h3.types';

const variants = variant({
    prop: 'styling',
    variants: {
        base: {
            color: 'text',
            fontSize: 'fontSizes.5',
        },
        get light() {
            return {
                ...this.base,
                fontWeight: 'normal',
            };
        },
    },
} as VariantArgs);

export const H3: H3StyledProps = styled.h3`
    ${variants}
    ${color}
    ${typography}
    ${space}
`;

export default H3;
