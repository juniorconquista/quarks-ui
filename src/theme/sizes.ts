export const breakpoints = ['0rem', '37.5rem', '62rem']

export const space = {
  none: '0',
  xxxs: '0.285rem',
  xxs: '0.571rem',
  xs: '0.857rem',
  sm: '1.142rem',
  md: '1.714rem',
  lg: '2.5rem',
  xl: '3.428rem',
  xxl: '6.857rem'
}

export const fontSizes = [
  '0.5rem', // 8px
  '0.67rem', // 10.72px - H6 / SMALL
  '0.83rem', // 13.28px - H5
  '0.875rem', // 14px
  '1rem', // 16px - H4/P/SPAN
  '1.17rem', // 18.72px - H3
  '1.5rem', // 24px - H2
  '1.8rem', // 28.8px
  '2rem' // 32px - H1
]

export const buttonSizes = {
  medium: {
    paddingTop: 8,
    paddingBottom: 8,
    paddingRight: 16,
    paddingLeft: 16,
    borderRadius: 16
  },
  small: {
    paddingTop: 4,
    paddingBottom: 4,
    paddingRight: 16,
    paddingLeft: 16,
    borderRadius: 16
  }
}

export const radii = {
  none: '0',
  sm: '0.285rem',
  md: '2.25rem',
  lg: '1.142rem',
  xl: '1.714rem'
}
