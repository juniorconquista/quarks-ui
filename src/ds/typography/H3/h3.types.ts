import { DefaultStyleProps } from '../../theme/types';
import { TypographyProps } from 'styled-system';

export type H3StyledProps = DefaultStyleProps &
    TypographyProps & {
        styling: 'base' | 'light';
    };

export interface H3Props {}
