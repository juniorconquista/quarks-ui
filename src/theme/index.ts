import colors from './colors';
import { breakpoints, space, fontSizes, radii } from './sizes';

const theme = {
    fontSizes,
    space,
    breakpoints,
    radii,
    colors: colors(),
};

//@ts-ignore
theme.breakpoints.mobile = breakpoints[0];
//@ts-ignore
theme.breakpoints.tablet = breakpoints[1];
//@ts-ignore
theme.breakpoints.desktop = breakpoints[2];

export default theme;
