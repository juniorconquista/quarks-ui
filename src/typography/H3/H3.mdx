---
name: H3
menu: Typography
---

import { Playground, Props } from 'docz';
import H3 from './H3';

# H3

Search engines use the headings to index the structure and content of your web pages. Users often skim a page by its headings. It is important to use headings to show the document structure.

Each HTML heading has a default size. However, you can specify the size for any heading with the style attribute, using the CSS _font-size_ property.

### Best practices

-   Use HTML headings for headings only. Don't use headings to make text BIG or bold.

## Properties

<Props of={H3} />

## Basic usage

```javascript
import { H3 } from 'quarks-ui/typography';
```

<Playground>
    <H3 styling="base">Title</H3>
</Playground>

## Customization

### Color

The color utility parses a component's _color_ and _bg_ props and converts them into CSS declarations. By default the raw value of the prop is returned. Color palettes can be configured with the ThemeProvider to use keys as prop values, with support for dot notation.

You can see all the color customization [here](https://styled-system.com/api/#color).

<Playground>
    <H3 styling="base" color="blue">
        Title
    </H3>
</Playground>

```javascript
// picks the value defined in `theme.colors.blue`
<H3 styling="base" color='blue'>Title</H3>

// picks up a nested color value using dot notation `theme.colors.gray[0]`
<H3 color='gray.0'>Title</H3>

// raw CSS color value
<H3 color='#f00'>Title</H3>

// background colors
<H3 bg='blue'>Title</H3>

// verbose prop
<H3 backgroundColor='blue'>Title</H3>
```

### Space

The space utility converts shorthand margin and padding props to _margin_ and _padding_ CSS declarations.

You can see all the space customization [here](https://styled-system.com/api#space).

<Playground>
    <H3 styling="base" m={2}>
        Title
    </H3>
</Playground>

```javascript
// sets margin value of `theme.space[2]`
<H3 styling="base" m={2}>Title</H3>

// sets a margin value of `16px` since it's greater than `theme.space.length`
<H3 m={16}>Title</H3>

// sets margin `'auto'`
<H3 m='auto'>Title</H3>

// sets margin `8px` fom mobile devices, `16px` for tablet and `32px` for desktop from the first breakpoint and up
<H3 m={{ mobile:2, tablet:3, desktop:4 }}>Title</H3>
```

### Typography

Typography props include _fontFamily_, _fontSize_, _fontWeight_, _lineHeight_, _letterSpacing_, _textAlign_, and _fontStyle_.

You can see all the typography customization [here](https://styled-system.com/api/#typography).

<Playground>
    <H3 styling="base" fontSize={5}>
        Title
    </H3>
</Playground>

```javascript
// font-size of `theme.fontSizes[5]`
<H3 styling="base" fontSize={5}>Title</H3>

// font-size `32px`
<H3 fontSize={32}>Title</H3>

// font-size `'2em'`
<H3 fontSize='2em'>Title</H3>

// font-size `10px` on all viewports and `12px` from the first breakpoint and up
<H3 fontSize={{ mobile:10, desktop:12 }}>Title</H3>

// fontFamily
<H3 fontFamily='mono'>Title</H3>

// textAlign
<H3 textAlign='center'>Title</H3>
<H3 textAlign={{ mobile: 'center', tablet: 'left', desktop: 'center' }}>Title</H3>

// lineHeight
<H3 lineHeight='1.25'>Title</H3>

// fontWeight
<H3 fontWeight='bold'>Title</H3>

// letterSpacing
<H3 letterSpacing='0.1em'>Title</H3>
```
