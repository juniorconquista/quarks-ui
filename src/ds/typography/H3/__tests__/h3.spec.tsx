import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

import H3 from '../H3';
import theme from '../../../theme';

describe('H3', () => {
    test('basic snapshot', () => {
        const tree = renderer
            .create(
                <ThemeProvider theme={theme}>
                    <H3 styling="base" />
                </ThemeProvider>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('should inherit the styling properties of the base variant', () => {
        const tree = renderer
            .create(
                <ThemeProvider theme={theme}>
                    <H3 styling="base" />
                </ThemeProvider>,
            )
            .toJSON();
        expect(tree).toHaveStyleRule('color', theme.colors.text);
        expect(tree).toHaveStyleRule('font-size', theme.fontSizes[5]);
    });

    test('should inherit the styling properties of the light variant', () => {
        const tree = renderer
            .create(
                <ThemeProvider theme={theme}>
                    <H3 styling="light" />
                </ThemeProvider>,
            )
            .toJSON();
        expect(tree).toHaveStyleRule('color', theme.colors.text);
        expect(tree).toHaveStyleRule('font-size', theme.fontSizes[5]);
        expect(tree).toHaveStyleRule('font-weight', 'normal');
    });

    test('should apply custom styling properties', () => {
        const tree = renderer
            .create(<H3 styling="base" mt="20px" p="3px" />)
            .toJSON();
        expect(tree).toHaveStyleRule('margin-top', '20px');
        expect(tree).toHaveStyleRule('padding', '3px');
    });
});
