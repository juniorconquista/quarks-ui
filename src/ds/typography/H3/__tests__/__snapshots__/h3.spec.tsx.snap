// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H3 basic snapshot 1`] = `
.c0 {
  color: #1B2b3a;
  font-size: 1.17rem;
}

<h3
  className="c0"
/>
`;
